File: lib/models/signalement.dart
class Signalement {
  String prenom;
  String nom;
  String telephone;
  String objet;
  String message_signalement;

  Signalement({
    required this.prenom,
    required this.nom,
    required this.telephone,
    required this.objet,
    required this.message_signalement,
  });

  factory Signalement.fromJson(Map<String, dynamic> json) {
    return Signalement(
      prenom: json['prenom'] as String,
      nom: json['nom'] as String,
      telephone: json['telephone'] as String,
      objet: json['objet'] as String,
      message_signalement: json['message_signalement'] as String,
    );
  }

  Map toMap() {
    return {
      "prenom": prenom,
      "nom": nom,
      "telephone": telephone,
      "objet": objet,
      "message_signalement": message_signalement,
    };
  }
}

File: lib/php/signalement.php
<?php
require 'connection.php';

$post = json_decode($_POST['data']);
$prenom = $post->prenom;
$nom = $post->nom;
//$telephone = $post->telephone;
$telephone = '77 111 11 11';
$objet = $post->objet;
$message_signalement = $post->message_signalement;
echo $telephone;
$msg = '';
$isok = false;


try {
    $req = $db->prepare("INSERT INTO signalement (`id`, `prenom`, `nom`, `telephone`, `objet`, `message_signalement`, `date_signalement`) VALUES (null,?,?,?,?,?, current_timestamp())");
    $req->execute(array($prenom, $nom, $telephone, $objet, $message_signalement));

    if ($req) {
        $msg = "Signalement envoyé";
        $isok = true;
    } else {
        $msg = "echec de l'envoi";
    }
} catch (\Throwable $th) {
    $msg = 'Erreur: ' . $th->getMessage();
}

echo json_encode([
    "data" => [
        $isok,
        $msg,
    ]
]);